src/api/WordsAPI.ts

```diff
@@ -34,3 +36,65 @@ export const useSearch = (word: string | undefined) => {
     enabled: !!word,
   });
 };
+
+const getDirectSearchResult = async (word: string | undefined) => {
+  if (!word) {
+    return [];
+  }
+  const dynamicData = await firstValueFrom(
+    SearchResource.query(getSearchQuery(word))
+  );
+
+  // Materialize data to actual JS objects so that we can manipulate it with lodash
+  const data = dynamicData.map((item) => {
+    const {
+      "@id": uri,
+      label,
+      vocabulary,
+      score,
+      snippetField,
+      snippetText,
+    } = item;
+    return {
+      uri,
+      label,
+      vocabulary,
+      score,
+      snippetField,
+      snippetText,
+    };
+  });
+
+  // Groups results with the same label
+  // adds isWord flag when there are multiple
+  const result = _(data)
+    .groupBy("label")
+    .map((objs, key) => {
+      return {
+        label: key,
+        // include text with highlighted search string, should be the same for all objects with the same label
+        displayText:
+          objs[0].snippetField === "label" ? objs[0].snippetText : key,
+        total_score: _.maxBy(objs, "score")?.score,
+        items: [...objs],
+        isWord: objs.length !== 1,
+        vocabularies: _.map(_.uniqBy(objs, "vocabulary"), "vocabulary"),
+      };
+    })
+    .orderBy("total_score", "desc")
+    .value();
+
+  return result;
+};
+
+export type DirectSearchResult = ReturnType<
```

Comment:
This is just a type helper to get a return type of the function above, minus the promise.